docs/src/cookbooks/two-side-column.md

```diff
@@ -0,0 +1,888 @@
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```
+</td>
+<td>
+
+
+```scala mdoc:silent
+class CaseStatementEnumModule1 extends Module {
+  
+  object AluMux1Sel extends ChiselEnum {
+    val selectRS1, selectPC = Value
+  }
+  
+   import AluMux1Sel._
+   val rs1, pc = IO(Input(UInt(3.W)))
+   val sel = IO(Input(AluMux1Sel()))
+   val out = IO(Output(UInt(3.W)))
+   
+   // default goes first
+   out := 0.U
+    
+   switch (sel) {
+     is (selectRS1) { out := rs1 }
+     is (selectPC)  { out := pc }
+  }
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new CaseStatementEnumModule1)
+```
+</td>
+         </tr>
+  <tr>
+<td>
+
+```
+module CaseStatementEnumModule2 (input clk);
+ 
+  typedef enum {INIT, IDLE, START, READY} StateValue;
+    StateValue state;
+    
+    
+ 
+    always @(posedge clk) begin
+        case (state)
+            IDLE    : state = START;
+            START   : state = READY;
+            READY   : state = IDLE ;
+            default : state = IDLE ;
+        endcase
+    end
+    
+endmodule
+```
+</td>
+<td>
+
+```scala mdoc:silent
+class CaseStatementEnumModule2 extends Module {
+  
+  object StateValue extends ChiselEnum {
+   val INIT  = Value(0x03.U) 
+    val IDLE  = Value(0x13.U) 
+    val START = Value(0x17.U) 
+    val READY = Value(0x23.U) 
+  }
+   import StateValues._
+    val state = RegInit(INIT)
+  val nextState = IO(Output(StateValue()))
+  nextState := state
+
+  
+  switch (state) {
+  is (INIT) {
+   state := IDLE   
+  }
+  is (IDLE) {  
+   state := START
+  }
+  is (START) { 
+   state := READY
+  }
+ is (READY) { 
+   state := IDLE
+  }
+  }
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new CaseStatementEnumModule2)
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+<!--
+# SystemVerilog Interfaces
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```
+module MyInterfaceModule(
+  input        clock,
+  input        reset,
+  output       io_in_ready,
+  input        io_in_valid,
+  input  [7:0] io_in_bits,
+  input        io_out_ready,
+  output       io_out_valid,
+  output [7:0] io_out_bits
+);
+  assign io_in_ready = io_out_ready; // @[main.scala 17:12]
+  assign io_out_valid = io_in_valid; // @[main.scala 17:12]
+  assign io_out_bits = io_in_bits; // @[main.scala 17:12]
+endmodule
+```
+</td>
+<td>
+
+```scala mdoc:silent
+class MyInterfaceModule extends Module {
+val io = IO(new Bundle {
+val in = Flipped(DecoupledIO(UInt(8.W)))
+val out = DecoupledIO(UInt(8.W))
+})
+
+val tmp = Wire(DecoupledIO(UInt(8.W)))
+tmp <> io.in
+io.out <> tmp
+io.out <> io.in
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new MyInterfaceModule)
+```
+
+</td>
+         </tr>
+    </table>
+
+<html>
+<body>
+-->
+
+
+# Multi-Dimensional Memories
+
```

Comment:
These examples are a bit swapped around, here's what needs to happen:
<img width="1016" alt="Screen Shot 2022-02-14 at 9 11 23 AM" src="https://user-images.githubusercontent.com/17858596/153912615-d12abecd-f169-4b1d-b579-724042232a5a.png">